src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -95,14 +97,16 @@ export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFo
 
       const nameParameterExists = isNameParameterExists(template);
 
-      const templateToProcess = produce(template, (draftTemplate) => {
+      const templateToProcess = produce(template, (draftTemplate: V1Template) => {
         if (nameParameterExists)
           replaceTemplateParameterValue(draftTemplate, NAME_INPUT_FIELD, vmName);
 
         const vmObject = getTemplateVirtualMachineObject(draftTemplate);
 
         if (!isEmpty(authorizedSSHKey)) {
-          draftTemplate.objects = template.objects.filter((obj) => obj?.kind !== SecretModel.kind);
+          draftTemplate.objects = template.objects.filter(
+            (obj: { kind: any }) => obj?.kind !== SecretModel.kind,
```

Comment:
Why use any? obj probably can be ResourceCommon

Reply:
Because I cannot know which type could be even better there. I am not familiar with this codebase. I am just improving it. Better than nothing.

Reply:
you can simply try, if it wont be the right type ts will complain, ts is scanning all code base with every save